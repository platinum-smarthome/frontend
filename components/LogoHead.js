import React, { Component } from 'react'
import { View, Image, StyleSheet } from 'react-native'

export default class LogoHead extends Component {
  render() {
    return (
      <View style={{alignSelf: 'center', paddingTop: 6 }}>
        <Image style={styles.logo} source={require('../components/assets/fortress_logo.png')}/>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  logo: {
    flex: 1,
    height: null,
    width: 150,
    resizeMode: 'contain'
  }
})