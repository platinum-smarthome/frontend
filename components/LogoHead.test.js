import React, { Component } from 'react'
import enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'

import LogoHead from './LogoHead'
import { View, Image } from 'react-native'

enzyme.configure({ adapter: new Adapter()})

let wrapper

beforeEach(() => {
  wrapper = shallow(<LogoHead status={true}/>)
})

describe('<LogoHead/>', () => {
  it('should render <LogoHead/>', () => {
    expect(wrapper).toBeDefined()
  })
})

describe('<LogoHead/> child is rendered', () => {
  it('Should render <View/>', () => {
    expect(wrapper.find(<View/>)).toBeDefined()
    expect(wrapper.find('View').length).toBe(1)  
  })
  it('Should render <Image/>', () => {
    expect(wrapper.find(<Image/>)).toBeDefined()
    expect(wrapper.find('Image').length).toBe(1)  
  })
})

describe('<Image/> renders the correct image', () => {
  it('should render the proper image if status is true', () => {
    expect(wrapper.find(<Image source={require('../components/assets/fortress_logo.png')}/>)).toBeDefined()
  })
})

describe('snapshot testing', () => {
  it('snapshot', () => {
    const tree = renderer.create(<LogoHead/>)
    expect(tree).toMatchSnapshot()
  })
})