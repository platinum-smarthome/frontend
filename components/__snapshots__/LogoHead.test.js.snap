// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing snapshot 1`] = `
<View
  style={
    Object {
      "flexGrow": 1,
      "paddingTop": 6,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../components/assets/fortress_logo.png",
      }
    }
    style={
      Object {
        "alignSelf": "center",
        "flex": 1,
        "height": null,
        "resizeMode": "contain",
        "width": 150,
      }
    }
  />
</View>
`;
